rllib/core/models/torch/encoder.py

```diff
@@ -189,7 +190,7 @@ def get_initial_state(self):
 
     @override(Model)
     def _forward(self, inputs: NestedDict, **kwargs) -> NestedDict:
-        x = inputs[SampleBatch.OBS]
+        x = inputs[SampleBatch.OBS].float()
```

Comment:
this doesn't feel right.
weights can be float16, while input OBS may be int32s ...

Reply:
I'd agree, but we did this in VisionNet forever now. Does torch barf if you choose lower resolution weights?

Reply:
We can convert with something like 
`x = inputs[SampleBatch.OBS].type(self.net.input_dtype)`.

Reply:
Moved the conversions into the Model objects that also create the layers. These layers are float32 and not configurable, but if anyone would overwrite that code, they'd be able to change the precision there.